core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -251,12 +255,20 @@ object ConsoleProducer {
     val options = parser.parse(args : _*)
     if(args.length == 0)
       CommandLineUtils.printUsageAndDie(parser, "Read data from standard input and publish it to Kafka.")
-    CommandLineUtils.checkRequiredArgs(parser, options, topicOpt, brokerListOpt)
 
     val useOldProducer = options.has(useOldProducerOpt)
+    val bootstrapServer = List(bootstrapServerOpt, brokerListOpt).flatMap(x => Option(options.valueOf(x))).head
```

Comment:
This should be done after we validate arguments. Otherwise, we can get a NoSuchElementException if neither `bootstrap-server` or `broker-list` is passed by the user. Same for other classes. Worth adding a test for this.